types/dec_coin.go

```diff
@@ -518,8 +518,15 @@ func (coins DecCoins) IsValid() bool {
 		}
 
 		lowDenom := coins[0].Denom
+		seenDenoms := make(map[string]bool)
+		seenDenoms[strings.ToUpper(lowDenom)] = true
+
 		for _, coin := range coins[1:] {
-			if strings.ToLower(coin.Denom) != coin.Denom {
+			denomUpper := strings.ToUpper(coin.Denom)
```

Comment:
Looking at the `IsValid` function, this should no longer be here? `"atom" != "ATOM"`?